### PR TITLE
[GlobalPropertyOpt] Simplify the code a little bit. NFCI.

### PR DESCRIPTION
**Pull request by dcci**
Use getFunction() instead of recursive calls to getParent() and
return early.

### CODE REVIEW

**Comment by dcci** [comment]
@swift-ci please smoke test

**Comment by dcci** [comment]
Passes the test. Can I merge this?

**Comment by slavapestov** [comment]
@dcci Do you have commit access now?

**Comment by dcci** [comment]
FWIW, I had since January 2016 :) [after I ported swift to FreeBSD] but then life took over so I wasn't able to contribute for a while :( Glad things changed recently.

**Comment by slavapestov** [comment]
Oh, cool. In the future, I think you should feel free to merge small cleanups like this one yourself.

**Comment by dcci** [comment]
Thanks for letting me know :) Hopefully we can fix the outstanding issue in clang and I can resume my work on better FreeBSD support [although I'm not holding my breath, as it seems a little tricky] :)

**Comment by dcci** [comment]
For reference https://bugs.llvm.org/show_bug.cgi?id=31620

File: lib/SILOptimizer/IPO/GlobalPropertyOpt.cpp [hunk 1]
```diff
@@ -68,18 +68,15 @@ class GlobalPropertyOpt {
...
-      } else {
-        os << entry.Value;
-      }
-      return os;
+      if (entry.Field)
+        return os << "field " << entry.Field->getName() << '\n';
+      if (!entry.Value)
+        return os << "unknown-address\n";
+      if (auto *Inst = dyn_cast<SILInstruction>(entry.Value))
+        os << Inst->getFunction()->getName() << ": " << entry.Value;
```
Comment (eeckstein) [inline lib/SILOptimizer/IPO/GlobalPropertyOpt.cpp]: There is a return missing here

**Comment by eeckstein** [comment]
@dcci Can you please fix the missing return in a follow-up commit?

**Comment by dcci** [comment]
Snap, I missed this one. Let me fix that.

**Pull request merged**